src/message.py

```diff
@@ -497,45 +524,74 @@ def modify_createfirm_exists_failure(firm_name):
 Only the CEO can do that.
 """
 
+not_ceo_or_coo_org = """
+Only the CEO or COO can do that.
+"""
+
 not_ceo_or_exec_org = """
-Only the CEO and executives can do that.
+Only a member of the board or an executive can do that.
+"""
+
+not_assoc_org = """
+Floor Traders cannot send invites.
 """
 
 promote_failure_org = """
 Couldn't promote user, make sure you used the correct username.
 """
 
-promote_full_org = """
-Could not promote this employee, since the firm is at its maximum executive limit.
-**Number of execs:** %EXECS%
+promote_coo_full_org = """
+Could not promote this employee since the firm already has a COO.
+"""
+
+promote_cfo_full_org = """
+Could not promote this employee since the firm already has a CFO.
+"""
+
+promote_execs_full_org = """
+Could not promote this employee since the firm is at its maximum executive limit.
+**Number of executives:** %EXECS%
 **Firm level:** %LEVEL%
 
-The CEO of the firm can raise this limit by upgrading with `!upgrade`.
+The CEO or CFO of the firm can raise this limit by upgrading with `!upgrade`.
 """
 
-def modify_promote_full(firm):
-    return promote_full_org.\
+def modify_promote_execs_full(firm):
+    return promote_execs_full_org.\
         replace("%EXECS%", str(firm.execs)).\
         replace("%LEVEL%", str(firm.rank + 1))
 
+promote_assocs_full_org = """
+Could not promote this employee since the firm is at its maximum associate limit.
+**Number of associates:** %ASSOCS%
+**Firm level:** %LEVEL%
+
+The CEO or CFO of the firm can raise this limit by upgrading with `!upgrade`.
+"""
+
+def modify_promote_assocs_full(firm):
+    return promote_assocs_full_org.\
+        replace("%ASSOCS%", str(firm.assocs)).\
+        replace("%LEVEL%", str(firm.rank + 1))
```

Comment:
Lines 559 through 575: Changed "!promote" error messages to reflect difference between associate limit and executive limit